doc: release-notes-3.7.rst: STM32 additions (#75325)
zephyrproject-rtos/zephyr: doc/releases/release-notes-3.7.rst

* Added support for these SoC series:

  * Added support for Ambiq Apollo3 Blue and Apollo3 Blue Plus SoC series.

>   * Added support for STM32H7R/S SoC series

Reviewer: ```suggestion
  * Added support for STM32H7R/S SoC series.
```

  * Added support for :ref:`Mikroe RA4M1 Clicker board <mikroe_clicker_ra4m1>`: ``mikroe_clicker_ra4m1``.
  * Added support for :ref:`Arduino UNO R4 WiFi board <arduino_uno_r4>`: ``arduino_uno_r4_wifi``.
  * Added support for :ref:`Renesas EK-RA8M1 board <ek_ra8m1>`: ``ek_ra8m1``.
  * Added support for :ref:`ST Nucleo H533RE <nucleo_h533re_board>`: ``nucleo_h533re``.
  * Added support for :ref:`ST STM32C0116-DK Discovery Kit <stm32c0116_dk_board>`: ``stm32c0116_dk``.

>   * Added support for :ref:`ST STM32H745I Discovery <stm32h745i_disco_board>`: ``stm32h745i_disco``.

Reviewer 2: should the  stm32h745i_disco` be precised with _/m4_ or _/m7_

Author: We're referring to boards rather than cores.

  * Added support for :ref:`ST STM32H7S78-DK Discovery <stm32h7s78_dk_board>`: ``stm32h7s78_dk``.
  * Added support for :ref:`ST STM32L152CDISCOVERY board <stm32l1_disco_board>`: ``stm32l152c_disco``.
  * Added support for :ref:`ST STEVAL STWINBX1 Development kit <steval_stwinbx1_board>`: ``steval_stwinbx1``.

* Added support for these Xtensa boards:

* Made these changes for ARM boards:

  * On :ref:`ST STM32H7B3I Discovery Kit <stm32h7b3i_dk_board>`: ``stm32h7b3i_dk_board``,
    enabled full cache management, Chrom-ART, double frame buffer and full refresh for
    optimal LVGL perfomances.

>   * On ST STM32 boards, stm32cubeprogrammer runner could now be used to program external
>     flash using ``--extload`` option.

Reviewer 3: ```suggestion
  * On ST STM32 boards, stm32cubeprogrammer runner can now be used to program external
    flash using ``--extload`` option.
```

* ADC

>   * Added support for STM32H7RS series.

Reviewer 3: ```suggestion
  * Added support for STM32H7R/S series.
```

* Battery backed up RAM

>   * Added support on STM32G0 and STM32H5 series.

Reviewer 3: For consistency with other messages:
```suggestion
  * Added support for STM32G0 and STM32H5 series.
```

* Clock control

  * Added support for Microcontroller Clock Output (MCO) on STM32H5 series.
  * Added support for MSI clock on STM32WL series.

* Counter

  * Added support for Ambiq Apollo3 series.

>   * Added support for STM32H7RS series.

Reviewer 3: ```suggestion
  * Added support for STM32H7R/S series.
```

* Entropy

>   * Added support for STM32H7RS series.

Reviewer 3: For consistency:
```suggestion
  * Added support for STM32H7R/S series.
```

  * Driver nxp_enet is no longer experimental.
  * All boards and SOCs with :dtcompatible:`nxp,kinetis-ethernet` compatible nodes
    reworked to use the new :dtcompatible:`nxp,enet` binding.

>   * Added support for PTP on STM32 compatible series (STM32F7, STM32H5 and STM32H7).

Reviewer 3: Nit:
```suggestion
  * Added support for PTP on compatible STM32 series (STM32F7, STM32H5 and STM32H7).
```

  * Added the c:func:`flash_fill` utility function which allows to write
    a single value across a provided range in a selected device.
  * Added support for RRAM on nrf54l15 devices.
  * Added support of non busy wait polling in STM32 OSPI driver.
  * Added support for STM32 XSPI external NOR flash driver (:dtcompatible:`st,stm32-xspi-nor`).

>   * Added support for XIP on external NOR flash on STM32 OSPI, QSPI and XSPI driver.

Reviewer 3: Nit:
```suggestion
  * Added support for XIP on external NOR flash in STM32 OSPI, QSPI and XSPI driver.
```

>   * STM32 OSPI driver: clk, dqs, ncs ports can now be configured by device tree configurable,
>     see :dtcompatible:`st,stm32-ospi`.

Reviewer 3: Nit:
```suggestion
  * STM32 OSPI driver: clk, dqs, ncs ports can now be configured by device tree configurable (see :dtcompatible:`st,stm32-ospi`)
```

* GNSS

* GPIO

  * Added support for Ambiq Apollo3 series.
  * Added Broadcom Set-top box(brcmstb) SoC GPIO driver.

>   * Added c:macro:`STM32_GPIO_WKUP` flag which allows to configure a pin as wakeup source from
>     Power Off state on STM32 family.

Reviewer 4: ```suggestion
  * Added c:macro:`STM32_GPIO_WKUP` flag which allows to configure specific pins as wakeup source from
    Power Off state on STM32 L4, U5, WB, & WL SoC series.
```

Author: Used `Added c:macro:`STM32_GPIO_WKUP` flag which allows to configure specific pins as wakeup source from
    Power Off state on STM32 series supporting :kconfig:option:`CONFIG_POWEROFF`.`

Reviewer 4: C0x supports Poweroff but does not support GPIO Wakeup pins yet.

* Hardware info

>   * Added device EUI64 ID support and implementation for STM32 series.

Author: Right, thanks!

* I2C

  * Added support for Ambiq Apollo3 series.

>   * On STM32 V2 driver, added support for a new :kconfig:option:`CONFIG_I2C_STM32_V2_TIMING`

Reviewer 3: Nit:
```suggestion
  * In STM32 V2 driver, added support for a new :kconfig:option:`CONFIG_I2C_STM32_V2_TIMING`
```

* PWM

>   * Added support for STM32H7RS series.

Reviewer 3: ```suggestion
  * Added support for STM32H7R/S series.
```

* RTC

  * Added Raspberry Pi Pico RTC driver.

>   * Added support for :kconfig:option:`CONFIG_RTC_ALARM` on all STM32 MCU series (except STM32F1)

Reviewer: ```suggestion
  * Added support for :kconfig:option:`CONFIG_RTC_ALARM` on all STM32 MCU series (except STM32F1).
```

* SMBUS

* SDHC

  * Added ESP32 SDHC driver (:dtcompatible:`espressif,esp32-sdhc`).
  * Added SDHC driver for Renesas MMC controller (:dtcompatible:`renesas,rcar-mmc`).

* Sensors

  * Added TMP114 driver
  * Added DS18S20 1-wire temperature sensor driver.
  * STM32 QDEC driver now supports encoder mode configuration (see :dtcompatible:`st,stm32-qdec`).
  * Added support for STM32 Digital Temperature Sensor (:dtcompatible:`st,stm32-digi-temp`).

* Serial

  * Added driver to support UART over Bluetooth LE using NUS (Nordic UART Service). This driver
    enables using Bluetooth as a transport to all the subsystems that are currently supported by
    UART (e.g: Console, Shell, Logging).

>   * Added DCache support in async DMA mode on STM32 driver.

Reviewer 4: ```suggestion
  * Added :kconfig:option:`CONFIG_NOCACHE_MEMORY` support in async DMA mode in STM32 UART driver.
```

  * Added support for STM32H7RS series.

* SPI

  * Added support for Ambiq Apollo3 series general IOM based SPI.
  * Added support for Ambiq Apollo3 BLEIF based SPI, which is specific for internal HCI.
  * Added support for :kconfig:option:`CONFIG_PM` and :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME` on STM32 SPI driver.

>   * Added support for DMA support with DCache enabled on STM32 SPI driver.

Reviewer 4: ```suggestion
  * Added support for :kconfig:option:`CONFIG_NOCACHE_MEMORY` in DMA SPI mode for STM32F7x SoC series.
```

  * Added support for STM32H7RS series.

* USB

* Video

>   * Added support for STM32 Digital camera interface (DCMI) driver (:dtcompatible:`st,stm32-dcmi`)

Reviewer: ```suggestion
  * Added support for STM32 Digital camera interface (DCMI) driver (:dtcompatible:`st,stm32-dcmi`).
```

* RTC

  * Added Raspberry Pi Pico RTC driver.
  * Added support for :kconfig:option:`CONFIG_RTC_ALARM` on all STM32 MCU series (except STM32F1).

* SMBUS

* SDHC

  * Added ESP32 SDHC driver (:dtcompatible:`espressif,esp32-sdhc`).
  * Added SDHC driver for Renesas MMC controller (:dtcompatible:`renesas,rcar-mmc`).

* Sensors

  * Added TMP114 driver
  * Added DS18S20 1-wire temperature sensor driver.
  * STM32 QDEC driver now supports encoder mode configuration (see :dtcompatible:`st,stm32-qdec`).
  * Added support for STM32 Digital Temperature Sensor (:dtcompatible:`st,stm32-digi-temp`).

* Serial

  * Added driver to support UART over Bluetooth LE using NUS (Nordic UART Service). This driver
    enables using Bluetooth as a transport to all the subsystems that are currently supported by
    UART (e.g: Console, Shell, Logging).
  * Added :kconfig:option:`CONFIG_NOCACHE_MEMORY` support in async DMA mode in STM32 driver.

>     Now it is possible to use UART in DMA mode, with :kconfig:option:`CONFIG_DCACHE` enabled
>     on STM32 F7 & H7 SoC series, as long as DMA buffers are placed in an uncached memory section.

Reviewer 3: Nit:
```suggestion
    It is now possible to use UART in DMA mode with :kconfig:option:`CONFIG_DCACHE` enabled
    on STM32 F7 & H7 SoC series, as long as DMA buffers are placed in an uncached memory section.
```

>   * Added support for STM32H7RS series.

Reviewer 3: ```suggestion
  * Added support for STM32H7R/S series.
```

* SPI

  * Added support for Ambiq Apollo3 series general IOM based SPI.
  * Added support for Ambiq Apollo3 BLEIF based SPI, which is specific for internal HCI.
  * Added support for :kconfig:option:`CONFIG_PM` and :kconfig:option:`CONFIG_PM_DEVICE_RUNTIME` on STM32 SPI driver.
  * Added support for :kconfig:option:`CONFIG_NOCACHE_MEMORY` in DMA SPI mode for STM32F7x SoC series.

>   * Added support for STM32H7RS series.

Reviewer 3: ```suggestion
  * Added support for STM32H7R/S series.
```

* USB

* Video

  * Added support for STM32 Digital camera interface (DCMI) driver (:dtcompatible:`st,stm32-dcmi`).

* W1

* Watchdog

  * Added :kconfig:option:`CONFIG_WDT_NPCX_WARNING_LEADING_TIME_MS` to set the leading warning time
    in milliseconds. Removed no longer used :kconfig:option:`CONFIG_WDT_NPCX_DELAY_CYCLES`.
  * Added support for Ambiq Apollo3 series.

>   * Added support for STM32H7RS series.

Reviewer 3: ```suggestion
  * Added support for STM32H7R/S series.
```

* STM32

  * Updated STM32F0 to cube version V1.11.5.
  * Updated STM32F3 to cube version V1.11.5.
  * Updated STM32F4 to cube version V1.28.0.
  * Updated STM32F7 to cube version V1.17.2.
  * Updated STM32G0 to cube version V1.6.2.
  * Updated STM32G4 to cube version V1.5.2.
  * Updated STM32H5 to cube version V1.2.0.
  * Updated STM32H7 to cube version V1.11.2.
  * Updated STM32L5 to cube version V1.5.1.
  * Updated STM32U5 to cube version V1.5.0.
  * Updated STM32WB to cube version V1.19.1.
  * Updated STM32WBA to cube version V1.3.1.

>   * Added STM32H7RS with cube version V1.0.0.

Reviewer 3: ```suggestion
  * Added STM32H7R/S with cube version V1.0.0.
```
